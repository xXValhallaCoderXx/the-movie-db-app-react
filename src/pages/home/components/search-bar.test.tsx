import React from "react";
import {shallow, mount} from "enzyme";
import SearchBar from "./search-bar";

describe("Search Bar Component", () => {
  const mockOnChange = jest.fn();
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, "useState");
  // @ts-ignore
  useStateSpy.mockImplementation(init => [init, setState]);
  it("should render correctly", () => {
    const component = shallow(<SearchBar onSearchChange={mockOnChange} />);
    expect(component).toMatchSnapshot();
  });
  it("should update state after input", () => {
    const wrapper = mount(<SearchBar onSearchChange={mockOnChange} />);
    const input = wrapper.find("input");
    // @ts-ignore
    input.instance().value = "Pokemon";
    input.simulate("change");
    expect(setState).toHaveBeenCalledWith("Pokemon");
  });
});
