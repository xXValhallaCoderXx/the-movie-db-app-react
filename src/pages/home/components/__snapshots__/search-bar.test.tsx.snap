// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar Component should render correctly 1`] = `
<Input
  onChange={[Function]}
  placeholder="Enter movie title..."
  value=""
/>
`;
